lib/msal-browser/src/cache/BrowserCacheManager.ts

```diff
@@ -400,26 +400,60 @@ export class BrowserCacheManager extends CacheManager {
      * Gets the active account
      */
     getActiveAccount(): AccountInfo | null {
-        const activeAccountIdKey = this.generateCacheKey(PersistentCacheKeys.ACTIVE_ACCOUNT);
-        const activeAccountId = this.browserStorage.getItem(activeAccountIdKey);
-        if (!activeAccountId) {
+        const activeAccountKeyNew = this.generateCacheKey(PersistentCacheKeys.ACTIVE_ACCOUNT_NEW);
+        const activeAccountValueNew = this.getItem(activeAccountKeyNew);
+        if (!activeAccountValueNew) {
+            const activeAccountKeyOld = this.generateCacheKey(PersistentCacheKeys.ACTIVE_ACCOUNT);
+            const activeAccountValueOld = this.getItem(activeAccountKeyOld);
+            if(!activeAccountValueOld)
+            {
+                return null;
+            }
+            const activeAccount = this.getAccountInfoByFilter({localAccountId: activeAccountValueOld})[0] || null;
+            if(activeAccount)
+            {
+                this.setActiveAccount(activeAccount);
+                this.browserStorage.removeItem(activeAccountKeyOld);
+                return activeAccount;
+            }
+            return null;
+        }
+        else
+        {
+            const activeAccountValueObj = this.validateAndParseJson(activeAccountValueNew);
+            if(activeAccountValueObj && activeAccountValueObj.hasOwnProperty("homeAccountId") && activeAccountValueObj.hasOwnProperty("localAccountId"))
+            {
+                return this.getAccountInfoByFilter({
+                    homeAccountId: activeAccountValueObj["homeAccountId"],
+                    localAccountId: activeAccountValueObj["localAccountId"]
+                })[0] || null;
+            }
             return null;
         }
-        return this.getAccountInfoByFilter({localAccountId: activeAccountId})[0] || null;
     }
 
     /**
      * Sets the active account's localAccountId in cache
      * @param account 
      */
     setActiveAccount(account: AccountInfo | null): void {
-        const activeAccountIdKey = this.generateCacheKey(PersistentCacheKeys.ACTIVE_ACCOUNT);
+        const activeAccountKey = this.generateCacheKey(PersistentCacheKeys.ACTIVE_ACCOUNT_NEW);
         if (account) {
             this.logger.verbose("setActiveAccount: Active account set");
-            this.browserStorage.setItem(activeAccountIdKey, account.localAccountId);
```

Comment:
Per comment above, keep this (instead of deleting on line 452), so that both the "old" and "new" values are in teh cache (and kept in sync) to ensure backwards and forwards compatibility.

Reply:
Updated with all of these comments! Thank you